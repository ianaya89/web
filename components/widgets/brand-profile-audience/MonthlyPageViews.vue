<template>
  <div>
    <h4>{{ $t('app.brand.profile.audience.page_views.chart.title') }}</h4>
    <div class="w-full mt-4">
      <line-chart :label-formatter="pageViewsLabelFormatter" :height="400" :series="pageViewsChartSeries" />
    </div>
  </div>
</template>

<script>
import LineChart from '@/components/charts/LineChart'
import { pageViewsLabelFormatter } from '@/services/chartFormatters'
import { BrandQuery } from '@/services/graphql'

export default {
  name: 'MonthlyPageViews',

  components: {
    LineChart
  },

  apollo: {
    brand: BrandQuery
  },

  props: {
    unitId: {
      type: String,
      required: true
    }
  },

  data: () => ({
    brand: BrandQuery.mock,
    pageViewsLabelFormatter: pageViewsLabelFormatter
  }),

  computed: {
    pageViewsChartSeries() {
      return {
        name: this.$t('app.brand.profile.audience.page_views.chart.title'),
        data: this.$get(this.brand, 'monthlyPageViews', []).map(item => {
          return {
            name: this.$filters.date(item.date, 'MMM'),
            date: item.date,
            unique: item.unique,
            y: item.total
          }
        })
      }
    }
  }
}
</script>

<style scoped>
</style>
