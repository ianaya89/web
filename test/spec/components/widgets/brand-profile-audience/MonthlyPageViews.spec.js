import { shallowMount } from '@vue/test-utils'
import MonthlyPageViews from '@/components/widgets/brand-profile-audience/MonthlyPageViews'

describe('MonthlyPageViews', () => {
  it('renders MonthlyPageViews', () => {
    const wrapper = shallowMount(MonthlyPageViews, {
      propsData: {
        unitId: 1
      }
    })

    wrapper.setData({
      pageViewsLabelFormatter: jest.fn()
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('returns page vews chart series', () => {
    const wrapper = shallowMount(MonthlyPageViews, {
      propsData: {
        unitId: 1
      },

      mocks: {
        $filters: {
          date: (date, format) => date + format
        }
      }
    })

    wrapper.setData({
      brand: {
        monthlyPageViews: [
          {
            date: '2019-08-08',
            unique: 4,
            total: 8
          }
        ]
      }
    })

    expect(wrapper.vm.pageViewsChartSeries.name).toEqual(
      'TRANSLATE:app.brand.profile.audience.page_views.chart.title'
    )
    expect(wrapper.vm.pageViewsChartSeries.data.length).toEqual(1)
    expect(wrapper.vm.pageViewsChartSeries.data[0].name).toEqual(
      '2019-08-08MMM'
    )
    expect(wrapper.vm.pageViewsChartSeries.data[0].date).toEqual('2019-08-08')
    expect(wrapper.vm.pageViewsChartSeries.data[0].unique).toEqual(4)
    expect(wrapper.vm.pageViewsChartSeries.data[0].y).toEqual(8)
  })
})
