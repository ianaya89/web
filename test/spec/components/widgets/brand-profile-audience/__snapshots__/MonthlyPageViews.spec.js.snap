// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyPageViews renders MonthlyPageViews 1`] = `
<div>
  <h4>
    TRANSLATE:app.brand.profile.audience.page_views.chart.title
  </h4>
   
  <div
    class="w-full mt-4"
  >
    <line-chart-stub
      enablemousetracking="true"
      height="400"
      labelformatter="vm => function () {
  return \`
            <div class=\\"text-h4\\">\${vm.$filters.date(this.point.date, 'MMM YYYY')}</div>
            <div class=\\"flex w-full\\">
              <div class=\\"w-1/2\\">
                <div class=\\"pt-3\\">
                  <div class=\\"text-grey text-p5 uppercase tracking-wide\\">
                    \${vm.$t('app.brand.profile.audience.page_views.chart.total')}
                  </div>
                  <div class=\\"text-h7 uppercase font-medium leading-tight\\">
                    \${vm.$filters.number(this.y)}
                  </div>
                </div>
              </div>
              <div class=\\"w-1/2\\">
                <div class=\\"pt-3\\">
                  <div class=\\"text-grey text-p5 uppercase tracking-wide\\">
                    \${vm.$t('app.brand.profile.audience.page_views.chart.unique')}
                  </div>
                  <div class=\\"text-h7 uppercase font-medium leading-tight\\">
                    \${vm.$filters.number(this.point.unique)}
                  </div>
                </div>
              </div>
            </div>
            \`;
}"
      series="[object Object]"
      tooltip="true"
    />
  </div>
</div>
`;
